### PR TITLE
Document k3s/MetalLB/helm proxy installation

### PR DESCRIPTION
**Pull request by cbosdo**
# Description

Documents an alternate way to deploy the Proxy containers on `k3s` using the `helm` chart

# Target branches

Which documentation version does this PR apply to?

- [X] Master (Default)
- [X] Manager-4.3
- [ ] Manager-4.2

### CODE REVIEW

File: modules/installation-and-upgrade/pages/proxy-container-k3s-installation.adoc [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+To configure the storage of the volumes to be used by the {productname} Proxy pod, define persistent volumes for the following claims.
+Please refer to the https://kubernetes.io/docs/concepts/storage/persistent-volumes/[[literal]``kubernetes``] or https://rancher.com/docs/k3s/latest/en/storage/[k3s] documentation for more details.
+The persistent volume claims are named:
+
+* [literal]``squid-cache-pv-claim``
+* [literal]``/package-cache-pv-claim``
+* [literal]``/tftp-boot-pv-claim``
+
+Create the configuration for the {productname} Proxy as documented in xref:proxy-container-setup.adoc[].
```
Comment (keichwa) [inline modules/installation-and-upgrade/pages/proxy-container-k3s-installation.adoc:114]: I rather vote to repeat (copy and paste) the configuration here again.  Or we could ask Ornela to make this a snippet and reuse it two times.

**Review by keichwa — CHANGES_REQUESTED** [review]

**Review by jcayouette — APPROVED** [review]
Looks good to me!

File: modules/installation-and-upgrade/pages/proxy-container-k3s-installation.adoc (hunk 1 above)
Comment (cbosdo) [reply modules/installation-and-upgrade/pages/proxy-container-k3s-installation.adoc:116]: I would rather have a snippet than duplicating the data as it makes it easier to maintain in the long run.

File: modules/installation-and-upgrade/pages/proxy-container-k3s-installation.adoc [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+----
+services:
+  annotations:
+    metallb.universe.tf/allow-shared-ip: key-to-share-ip
+    metallb.universe.tf/loadBalancerIPs: 192.168.122.241
+----
+
+To configure the storage of the volumes to be used by the {productname} Proxy pod, define persistent volumes for the following claims.
+Please refer to the https://kubernetes.io/docs/concepts/storage/persistent-volumes/[[literal]``kubernetes``] or https://rancher.com/docs/k3s/latest/en/storage/[k3s] documentation for more details.
```
[the remark this answers is not in the document]
Comment (cbosdo) [reply modules/installation-and-upgrade/pages/proxy-container-k3s-installation.adoc]: fixed

File: modules/installation-and-upgrade/pages/proxy-container-k3s-installation.adoc [hunk 3]
```diff
@@ -0,0 +1,129 @@
+[[installation-proxy-containers-k3s]]
+= Install Containerized {productname} Proxy on [literal]``k3s``
+
+[[installation-proxy-containers-k3s-k3s]]
+== Installing k3s
+
+On the container host machine, install [literal]``k3s`` without the load balancer and traefik router:
```
[the remark this answers is not in the document]
Comment (cbosdo) [reply modules/installation-and-upgrade/pages/proxy-container-k3s-installation.adoc]: done

File: modules/installation-and-upgrade/pages/proxy-container-k3s-installation.adoc [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+curl -sfL https://get.k3s.io | INSTALL_K3S_EXEC="--disable=traefik --disable=servicelb --tls-san=<K3S_HOST_FQDN>" sh -
+----
+
+[[installation-proxy-containers-k3s-cluster-access]]
+== Configuring cluster access
+
+[literal]``helm`` needs a configuration file to connect to the target kubernetes cluster.
+
+On the cluster server machine run the following command
+The resulting [path]``kubeconfig-k3s.yaml`` can be optionally transfered to a work machine:
```
[the remark this answers is not in the document]
Comment (cbosdo) [reply modules/installation-and-upgrade/pages/proxy-container-k3s-installation.adoc]: done

File: modules/installation-and-upgrade/pages/proxy-container-k3s-installation.adoc [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+
+----
+kubectl config view --flatten=true | sed 's/127.0.0.1/<K3S_HOST_FQDN>/' >kubeconfig-k3s.yaml
+----
+
+Before calling [literal]``helm``, run
```
[the remark this answers is not in the document]
Comment (cbosdo) [reply modules/installation-and-upgrade/pages/proxy-container-k3s-installation.adoc]: done

File: modules/installation-and-upgrade/pages/proxy-container-k3s-installation.adoc [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+----
+export KUBECONFIG=/path/to/kubeconfig-k3s.yaml
+----
+
+[[installation-proxy-containers-k3s-helm]]
+== Installing helm
+
+[NOTE]
+====
+The Containers Module is required to install `helm`.
```
[the remark this answers is not in the document]
Comment (cbosdo) [reply modules/installation-and-upgrade/pages/proxy-container-k3s-installation.adoc]: Fixed

File: modules/installation-and-upgrade/pages/proxy-container-k3s-installation.adoc [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+====
+
+Simply run:
+
+    zypper in helm
```
[the remark this answers is not in the document]
Comment (cbosdo) [reply modules/installation-and-upgrade/pages/proxy-container-k3s-installation.adoc]: done

File: modules/installation-and-upgrade/pages/proxy-container-k3s-installation.adoc [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+----
+helm repo add metallb https://metallb.github.io/metallb
+helm install --create-namespace -n metallb metallb metallb/metallb 
+----
+
+[literal]``MetalLB`` still requires a configuration to know the virtual IP address range to be used.
+In this example, the virtual IP addresses will be from [literal]``192.168.122.240`` to [literal]``192.168.122.250``, but that range could be lowered to a single address if the host only exposes the {productname} proxy.
+These addresses need to be a subset of the server network.
+
+Create a [path]``metallb-config.yaml`` with content like the following with an IP address range that aligns with the deployed network:
```
[the remark this answers is not in the document]
Comment (cbosdo) [reply modules/installation-and-upgrade/pages/proxy-container-k3s-installation.adoc]: done

File: modules/installation-and-upgrade/pages/proxy-container-k3s-installation.adoc [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+kubectl apply -f metallb-config.yaml
+----
+
+
+[[installation-proxy-containers-k3s-deploy]]
+== Deploying the {productname} proxy helm chart
+
+Create a configuration file forcing the IP address that [literal]``MetalLB`` will use for the {productname} Proxy services.
+This IP address needs to be the one to which the proxy FQDN entered when creating the proxy configuration.
+It also needs to be resolvable from both the {productname} server and the systems to connect to the proxy.
```
[the remark this answers is not in the document]
Comment (cbosdo) [reply modules/installation-and-upgrade/pages/proxy-container-k3s-installation.adoc]: done

**Comment by cbosdo** [comment]
@keichwa I addressed all of your comments but the last one

File: .changelog [hunk 10]
```diff
@@ -4,6 +4,7 @@
 # - Fixed error in Bat section of Upgrade Guide (bsc#1234567)
 # For guidelines: https://en.opensuse.org/openSUSE:Creating_a_changes_file_(RPM)#Changelog_section_.28.25changelog.29
 
+- Document helm deployment of the proxy on k3s and MetalLB
```
Comment (keichwa) [inline .changelog:7]:
```suggestion
- Document helm deployment of the proxy on k3s and MetalLB in Installation
and Upgrade Guide
```

File: modules/installation-and-upgrade/pages/proxy-container-k3s-installation.adoc (hunk 1 above)
Comment (keichwa) [reply modules/installation-and-upgrade/pages/proxy-container-k3s-installation.adoc:116]: ok, for the moment we can keep it as-is (imo).

**Review by keichwa — APPROVED** [review]
Thank you, I'll merge and backport it now.

**Pull request merged**